Formula/osc-cli.rb

```diff
@@ -0,0 +1,26 @@
+class OscCli < Formula
+  include Language::Python::Virtualenv
+
+  desc "Official Outscale CLI providing connectors to Outscale API"
+  homepage "https://github.com/outscale/osc-cli"
+  url "https://github.com/outscale/osc-cli/releases/download/v1.6.0/osc-sdk-1.6.tar.gz"
+  sha256 "6319af4ee3f2effd991e8bd0fcd2b49914cf200915a7b1f606957db8eeb14ed8"
+  license "BSD-3-Clause"
+
+  depends_on "python@3.9"
+
+  def install
+    venv = virtualenv_create(libexec, "python3")
+
+    system libexec/"bin/pip", "install", "-v", "fire",
+                                  "--ignore-installed", buildpath
+    system libexec/"bin/pip", "uninstall", "-y", "osc-sdk"
+    venv.pip_install_and_link buildpath
+  end
+
+  test do
+    # we test the help wich is printed in stderr :(
+    str = shell_output("#{bin}/osc-cli help 2>&1 >/dev/null ; true")
+    str.include?("SERVICE") == true
```

Comment:
You really shouldn't be testing help though.
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Do `assert_match` instead of using `include?`.

Reply:
A problem I have is that osc-cli use a configuration file that should be place in `~/home/.osc/config.json`, if not present I can't make a call on the API.
For next the osc-cli version we will fix it so we can pass the configuration path as a parameter of osc-cli.

Reply:
In the test block, you have access to `HOME` as `testpath`. So you can write a config file there.

Reply:
thanks, I'll make a better test then.